addons/website_sale/controllers/main.py

```diff
@@ -568,21 +568,21 @@ def checkout_form_validate(self, mode, all_form_values, data):
         if data.get("vat") and hasattr(Partner, "check_vat"):
             if data.get("country_id"):
                 data["vat"] = Partner.fix_eu_vat_number(data.get("country_id"), data.get("vat"))
-            partner_dummy = Partner.new({
-                'vat': data['vat'],
-                'country_id': (int(data['country_id'])
-                               if data.get('country_id') else False),
-            })
+            partner_dummy = Partner.new(self._get_vat_validation_fields(data))
             try:
                 partner_dummy.check_vat()
-            except ValidationError:
+            except ValidationError as exception:
                 error["vat"] = 'error'
+                error_message.append(exception.name)
```

Comment:
HI @jco-odoo and @ren-odoo 

I leave this comment open waiting for your feedback about functional review, if you have created this as a ticket to Odoo, or if you are considering that we remove completely showing the exception of the error in the VAT validation.

Best Regards

Reply:
@JKE-be Is it possible that right now, in the website when you give a faulty VAT number the system will keep the field red, but will not give an error with why it is wrong?  It is because by adding this it would at least be clear where the error comes from, but no idea how you see this.  (Would be nice if you could check this pr again)

Reply:
yes error["vat"] will make it red, error_message will be just display on top of checkout

ps: in v14, .name is depreacated if i'm not wrong